firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayPanelViewController.swift

```diff
@@ -162,9 +161,11 @@ class TabDisplayPanelViewController: UIViewController,
     }
 
     func newState(state: TabsPanelState) {
+        if state != tabsState {
```

Comment:
Yes i agree i'd be much more cleaner. The thing also for the other states i don't know if comparing the state and applying changes for different state could cause any side effects.
I believe it should be safe to do the comparison at the top since the state should be `Equatable` what do you think ?

Reply:
I mean, if we're comparing at the top of `newState` we should be fine, I think, because it's only doing this particular state. And yeah, everything should be `Equatable` state wise, so we should be safe.

But maybe I didn't understand the question. What do you mean by "also for the other states i don't know if comparing the state and applying changes for different state could cause any side effects"